ingestion/src/metadata/ingestion/source/database/datalake.py

```diff
@@ -0,0 +1,315 @@
+                columns=table_columns,
+            )
+            schema_entity = DatabaseSchema(
+                id=uuid.uuid4(),
+                name=bucket_name,
+                database=EntityReference(id=database_entity.id, type="database"),
+                service=EntityReference(id=self.service.id, type="databaseService"),
+            )
+            table_and_db = OMetaDatabaseAndTable(
+                table=table_entity,
+                database=database_entity,
+                database_schema=schema_entity,
+            )
+            try:
+                if self.source_config.generateSampleData:
+                    table_data = self.fetch_sample_data(df, bucket_name)
+                    if table_data:
+                        table_entity.sampleData = table_data
+            # Catch any errors during the ingestion and continue
+            except Exception as err:  # pylint: disable=broad-except
+                logger.error(repr(err))
+                logger.error(traceback.format_exc())
+                logger.error(err)
+
+            yield table_and_db
+
+        except Exception as err:
+            logger.debug(traceback.format_exc())
+            logger.error(traceback.format_exc())
+            logger.error(err)
+
+    def fetch_sample_data(self, df, table: str) -> Optional[TableData]:
+        try:
+            cols = []
+            table_columns = self.get_columns(df)
+
+            for col in table_columns:
+                cols.append(col.name.__root__)
+            table_rows = df.values.tolist()
+
+            return TableData(columns=cols, rows=table_rows)
+        # Catch any errors and continue the ingestion
+        except Exception as err:  # pylint: disable=broad-except
+            logger.debug(traceback.format_exc())
+            logger.error(f"Failed to generate sample data for {table} - {err}")
+        return None
+
+    def get_columns(self, df):
+        df_columns = list(df.columns)
+        for column in df_columns:
+            try:
+
+                if df[column].dtypes.name == "int64":
+                    data_type = "INT"
+                if df[column].dtypes.name == "object":
+                    data_type = "INT"
+                parsed_string = {}
+                parsed_string["dataTypeDisplay"] = column
+                parsed_string["dataType"] = data_type
+                parsed_string["name"] = column[:64]
+                parsed_string["dataLength"] = parsed_string.get("dataLength", 1)
+                yield Column(**parsed_string)
+            except Exception as err:
+                logger.debug(traceback.format_exc())
+                logger.error(traceback.format_exc())
+                logger.error(err)
+
+    def close(self):
+        pass
+
+    def get_status(self) -> SourceStatus:
+        return self.status
+
+    def test_connection(self) -> None:
+        pass
+
+    @staticmethod
+    def _get_bucket_name_with_prefix(bucket_name: str) -> str:
```

Comment:
functions only related to S3/GCS could be refactores to a s3_utils or gcs_utils modules